requiam/commons.py

```diff
@@ -31,6 +34,56 @@ def figshare_stem(stem='', production=True):
     return stem_query
 
 
+def dict_load(config_file, vargs=None):
+    """
+    Purpose:
+      Read in a config INI file using configparser and return a dictionary
+      with sections and options
+
+    :param config_file: str. Full/relative path of configuration file
+    :param vargs: Dictionary containing command-line arguments
+    :return config_dict: dict of dict with hierarchy of sections follow by options
+    """
+
+    config = configparser.ConfigParser()
+    config.read(config_file)
+
+    # Read in default settings
+    config_dict = {}
+    for section in config.sections():
+        config_dict[section] = {}
+        for option in config.options(section):
+            option_input = config.get(section, option)
+            if option_input in ['True', 'False']:
+                config_dict[section][option] = config.getboolean(section, option)
+            else:
+                if option_input.isdigit():  # set as integer
+                    config_dict[section][option] = int(option_input)
+                else:
+                    config_dict[section][option] = config.get(section, option)
+
+    # Populate with command-line arguments overrides
+    config_dict['extras'] = {}
+    if not isinstance(vargs, type(None)):
```

Comment:
What is the difference between `not isinstance(x, type(None))` and `not x`? If there is no difference I suggest using `not x`. here and in line 69.

Reply:
Fair question. PEP8 would say when comparing types, to do so with `isinstance`. In most cases `vargs` is a dictionary. I suppose the more correct approach is to check if the dictionary is empty.

Reply:
When checking for empty dictionaries, since empty dictionaries are falsy by default, I suggest using:
```python
if not vargs:
  pass
```

Reply:
I use a combination of None handling and empty dict